core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/conditional/CaseExpression.kt

```diff
@@ -0,0 +1,143 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.conditional
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.Resolvable
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.UnaliasedExpression
+import ch.ergon.dope.validtype.BooleanType
+import ch.ergon.dope.validtype.ValidType
+
+private const val CASE = "CASE"
+private const val WHEN = "WHEN"
+private const val THEN = "THEN"
+private const val END = "END"
+private const val ELSE = "ELSE"
+
+class CaseClass<T : ValidType>(val case: TypeExpression<T>) : Resolvable {
+    override fun toDopeQuery(): DopeQuery {
+        val caseDopeQuery = case.toDopeQuery()
+        return DopeQuery("$CASE ${caseDopeQuery.queryString}", caseDopeQuery.parameters)
+    }
+}
+
+class SimpleCaseExpression<T : ValidType, U : ValidType>(
+    val case: TypeExpression<T>,
+    val firstWhenThen: SearchResult<T, out U>,
+    vararg val additionalWhenThen: SearchResult<T, out U>,
+) : TypeExpression<U> {
+    override fun toDopeQuery(): DopeQuery {
+        val caseDopeQuery = case.toDopeQuery()
+        val whenThenDopeQueries =
+            listOf(firstWhenThen, *additionalWhenThen).map { it.searchExpression.toDopeQuery() to it.resultExpression.toDopeQuery() }
+        return DopeQuery(
+            queryString = "$CASE ${caseDopeQuery.queryString} " +
+                "${whenThenDopeQueries.joinToString(" ") { "$WHEN ${it.first.queryString} $THEN ${it.second.queryString}" }} " +
+                END,
+            parameters = caseDopeQuery.parameters + whenThenDopeQueries.fold(emptyMap()) { parameters, query ->
+                parameters + query.first.parameters + query.second.parameters
+            },
+        )
+    }
+}
+
+class SimpleElseCaseExpression<T : ValidType, U : ValidType>(
```

Comment:
Do we really need this additional class? Can't we make the elseCase-nullable?

Reply:
Yes we need this class, because the user could otherwise call `whenThen().otherwise().whenThen()` and as discussed we don't want that